### ISSUE TITLE
Sql image data type can't be mapped to SqlBinary in System.Data.SqlTypes

### ISSUE THREAD

**Issue by MaxAkbar** [dotnet/SqlClient#1573]

**Comment by JRahnama** [comment]
@MaxAkbar we will look into this and will get back to you soon.

**Comment by JRahnama** [comment]
@MaxAkbar Isn't that an expected behavior According to [this documentation](https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/sql-server-data-type-mappings?msclkid=88a54961b6b011ec9a1eea5b0295d226)?
SqlType `image` is mapped as `Byte[]`, However since SqlBinary has a fixed length you only can get the value by `GetSqlBinary`.

**Comment by MaxAkbar** [comment]
No, I don't think so. The fact that you can map SqlBinary from the DataReader tells me that it should be mapped as coded in my example. Also, if that is not true then how come all other types are mapped except this one.

image | Byte[] | [Binary](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqldbtype#system-data-sqldbtype-binary)| [GetSqlBinary](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqldatareader.getsqlbinary)| [SqlBinary](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqltypes.sqlbinary?view=net-6.0)| [GetBytes](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqldatareader.getbytes)
-- | -- | -- | -- | -- | --
Sql Server native type|ADO.NET Framework and Core (yes it can be used as a workaround) |This is a data type that is used when setting Command Parameters|The DataReader method returns a SqlBinary|C# representation of the Sql Type|Same as Byte[]

One more thing if you run my code and debug the following:

`DataTable dataTable = sqlRdr.GetSchemaTable();`

You will notice that the provider specific data type is SqlBinary. All the types are stated as the correct mapping.
![Sql Types Mapping](https://raw.githubusercontent.com/MaxAkbar/SharedImages/main/sqltypes.png)


I just looked at how the DataReader maps to SqlBinary and it looks like there is a workaround class that does it internally.

```
case TdsEnums.SQLIMAGE:
                    {
                        // Note: Better not come here with plp data!!
                        Debug.Assert(length <= TdsEnums.MAXSIZE);
                        byte[] b = new byte[length];
                        if (!stateObj.TryReadByteArray(b, length))
                        {
                            return false;
                        }
                        value.SqlBinary = SqlTypeWorkarounds.SqlBinaryCtor(b, true);

                        break;
                    }
```

https://github.com/dotnet/SqlClient/blob/e1ee48eded4d3c2b53b0f289045ecbb61167cc11/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs#L6280

**Comment by Wraith2** [comment]
The image text and ntext types are deprecated and may be removed in a future version. See https://learn.microsoft.com/en-us/sql/t-sql/data-types/ntext-text-and-image-transact-sql?view=sql-server-ver16

**Comment by ErikEJ** [comment]
@Wraith2 in reality is is never going to happen.

**Comment by Wraith2** [comment]
We'll see. However if you read the page you'll see that the recommendation for using image is to map it as varbinary(max) and in that case not cleanly mapping using image in this codebase makes some sense.
